### ISSUE TITLE
[FE] My VA: Unit test coverage investigation

### ISSUE THREAD

**Issue by ajakabcin** [department-of-veterans-affairs/va.gov-team#68659]
## Background
In our staging review for the dot indicator we received QA feedback that our unit test coverage did not meet QA standards. We need to investigate this and determine if we just didn't share with them all of our test coverage, or if there are areas we need to improve, and whether these are Profile tests, My VA tests or both so that we aren't blocked on future work as this is now a launch-blocking standard.

- QA feedback ticket #68208
- [ Platform QA standards](https://depo-platform-documentation.scrollhelp.site/developer-docs/quality-assurance-standards)
- [Unit Coverage Report (Platform)](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/frontend-support-dashboard/unit-test-coverage-report/) 

## Tasks
- [x] Review existing unit test coverage
- [x] Identify areas for improvement 
- [x] Determine wether these are profile or my va tests

## Acceptance Criteria
- [ ] We know if our unit test coverage is meeting the platform standard, and if it's not we have identified where the improvements are needed

## Validation
- _Who can validate this ticket? (FE, BE, Design, PM)?_  Dev
- _How can this work be validated?_

**Comment by allisonlu** [comment]
The percentages listed in the [QA ticket above](https://github.com/department-of-veterans-affairs/va.gov-team/issues/68208) only apply to My VA. Both Profile and My VA are listed as separate applications in [Platform's Unit Test Coverage Report](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/frontend-support-dashboard/unit-test-coverage-report/).

## current state of coverage
These are our coverage stats as of November 7th, 2023. (sorted by Statements least -> most % covered)

![dashboard-unit-test-coverage-nov-7-2023-sorted](https://github.com/department-of-veterans-affairs/va.gov-team/assets/8542413/5ca4eec7-01f6-4648-835d-e2021f6deb24)

## recommendations
- focus on increasing unit tests for the following files in `dashboard/utils` and `dasboard/actions`. (I think getting these over 80% will increase our totals to >70%)

| utils | actions | 
|-- |--|
| ![dashboard-utils-unit-test-coverage-nov-7-2023](https://github.com/department-of-veterans-affairs/va.gov-team/assets/8542413/4ef3c943-39f5-405c-bbe2-b38f53ff7320) | ![dashboard-actions-unit-test-coverage-nov-7-2023](https://github.com/department-of-veterans-affairs/va.gov-team/assets/8542413/950d14a1-28a3-47ce-8ec7-ec8cec1f34c6) |

- then turn our focus to `dashboard/tests/helpers.js` and `dashboard/components/education-and-training/EducationAndTraining.jsx`   

| tests | education-and-training | 
|-- |--|
| ![dashboard-tests-unit-test-coverage-nov-7-2023](https://github.com/department-of-veterans-affairs/va.gov-team/assets/8542413/72f583d5-6fdf-4110-93fc-ac661bf6e359) |  ![dashboard-edu-and-training-unit-test-coverage-nov-7-2023](https://github.com/department-of-veterans-affairs/va.gov-team/assets/8542413/3b30f12c-5071-4b11-aad4-9e886a6dba85)|

**Comment by allisonlu** [comment]
@ajakabcin 

^ let me know if this is enough information, and if we have tickets for these unit tests. If not, I can help create them!

**Comment by ajakabcin** [comment]
Amazing @allisonlu this is exactly what I needed thank you! I'll look at what we have already in terms of test tickets (we still have some from the backlog) and let you know. Thanks again! I'll close this out :)
